client/app/core/session.service.js

```diff
@@ -87,42 +91,56 @@ export function SessionFactory($http, $sessionStorage, gettextCatalog, $window,
   }
 
   function setRBAC(productFeatures) {
+    var deferred = $q.defer();
     RBAC.set(productFeatures);
-
     // FIXME move these to the RBAC service as well
-    $state.actionFeatures = setRBACForActions(productFeatures);
-    $state.navFeatures = setRBACForNavigation(productFeatures);
+
+    setRBACForActions(productFeatures).then(function (actionsRBAC) {
```

Comment:
Furthering the feature bleed of RBAC functionality isn't ideal (that lil fixme is on to something)